features/telemetry/registry/src/main/java/org/opennms/netmgt/telemetry/protocols/registry/impl/TelemetryServiceRegistryImpl.java

```diff
@@ -117,6 +119,10 @@ public T getService(BD beanDefinition) {
         final ServiceFactoryRegistration<F> registration = delegate.lookup(beanDefinition.getClassName(), null);
         if (registration != null) {
             final T service = (T) registration.getServiceFactory().createBean(beanDefinition);
+
+            final BeanWrapper wrapper = PropertyAccessorFactory.forBeanPropertyAccess(service);
```

Comment:
I think there may have been a reason why this was done in the individual factories rather than here... something something OSGi.

Reply:
No, I think it was just placed in the factories were it was currently used.